pkg/validation/app.go

```diff
@@ -256,6 +257,22 @@ func (v *Validator) validateKubeConfig(ctx context.Context, cr v1alpha1.App) err
 }
 
 func (v *Validator) validateLabels(ctx context.Context, cr v1alpha1.App) error {
+	// This is for migrations towards managing App CRs from the org namespace.
+	// For the transition time being we must retain backward compatibility for the
+	// App CRs in cluster namespaces.
+	isManagedInOrg := !key.InCluster(cr) && key.IsManagedInOrg(cr)
```

Comment:
For WC Apps (when `.spec.KubeConfig.InCluster: false`) and managed from organization namespace we use different logic for validating labels. Otherwise fallback to the current logic.

Reply:
I think this is fine. Apps such as `cluster-openstack` with `InCluster: true` are managed by the MC's `app-operator` so they need an `app-operator` version label of `0.0.0` like other in-cluster apps.